reflectable/lib/capability.dart

```diff
@@ -518,11 +518,11 @@ class _StringInvocation extends StringInvocation {
 /// doesn't have the capabilities to invoke it.
 class ReflectableNoSuchMethodError extends Error
     implements NoSuchCapabilityError {
-  final Object receiver;
+  final Object? receiver;
   final String memberName;
   final List positionalArguments;
-  final Map<Symbol, dynamic> namedArguments;
-  final List<Symbol> existingArgumentNames;
+  final Map<Symbol, dynamic>? namedArguments;
```

Comment:
Could we add a line of documentation explaining what it means for this to be null? Why not just an empty map?

Reply:
The implementation does handle a `namedArguments` whose value is null, and the only occurrences of null aware invocations (like `namedArguments?.keys`) existed already before this PR. Hence, allowing clients of reflectable to pass null for this argument is an existing feature. I don't see the need to take that feature away.

On the other hand, `_StringInvocation` does replace that null by an empty map, so the error messages won't talk about null. Added a dartdoc comment.